src/index.ts

```diff
@@ -1,3 +1,73 @@
-export const add = (a: number, b: number) => {
-  return a + b
+import { ApiPromise, WsProvider } from '@polkadot/api'
+import { options } from './options'
+
+/**
+ * The Mangata class defines the `getInstance` method that lets clients access the unique singleton instance. Design pattern Singleton Promise is used.
+ */
+export class Mangata {
+  private static instance: Mangata
+  private apiPromise: Promise<ApiPromise> | null
+  private uri: string
+
+  /**
+   * The Mangata's constructor is private to prevent direct
+   * construction calls with the `new` operator.
+   */
+  private constructor(uri: string) {
+    this.apiPromise = null
+    this.uri = uri
+  }
+
+  /**
+   * Initialised via isReady & new with specific provider
+   */
+  private async connect() {
+    if (!this.apiPromise) {
+      const provider = new WsProvider(this.uri)
+      this.apiPromise = new ApiPromise(options({ provider })).isReady
+    }
+
+    return this.apiPromise
+  }
+
+  /**
+   * The static method that controls the access to the Mangata instance.
+   */
+  public static getInstance(uri: string): Mangata {
+    if (!Mangata.instance) {
+      Mangata.instance = new Mangata(uri)
+    }
+
+    return Mangata.instance
+  }
+
+  /**
+   * Retrieve the chain name
+   */
+
+  public async getChain(): Promise<string> {
```

Comment:
Since in the nearest future we will have a lot of callable actions, it makes sense to have them in separate files. They could be classified by API flavor (rpc, tx, query)

Reply:
Yes exactly this three operation are only here for general purpose how class with methods will look like. I will move this method to separate files.

Reply:
Will we need to maintain definitions of all available methods here, or the definitions could be moved to corresponding folders/files?

I'm afraid of keeping all in a single place because of big amount of calls to be implemented and different nature of different API flavors- `query` retrieves state, `tx` modifies it etc